### PR TITLE
RHBA-502: [project-oriented] assets not indexed when repo cloned via …

### PR DESCRIPTION
**Pull request by mbarkley**
…rest

### CODE REVIEW

**Comment by mbarkley** [comment]
Related PR: https://github.com/kiegroup/kie-wb-common/pull/1485

**Comment by mbarkley** [comment]
This commit reintroduces the batch index job, but now it only runs when a new repository is created that contains files.

Consequently there are now two indexing scenarios:

1. A new repo with no files is created.
2. A new repo is created that is a clone of another, containing some files.

In scenario (1), the batch index job does not run, and indexing happens when a first file is added. This is ideal for loading examples (where we do not clone, but instead write the files in a single commit).

In scenario (2), the batch index job runs after the repository is created so that we index pre-existing files before any new assets are created.

@adrielparedes @porcelli @ederign Would any of you care to review this?

**Comment by mbarkley** [comment]
@manstis Hmm, thanks for bringing that up. This may be covered [here](https://github.com/kiegroup/appformer/pull/227/files#diff-6ae75720689ed79fa1dcbc2773880c73R182) but I will have to test this tomorrow.

**Comment by jhrcek** [comment]
Build failed because of some jenkins issue? ` hudson.remoting.ChannelClosedException: Remote call on ci-rhos-rhel7-xlarge-6633 failed. The channel is closing down or has closed down`
Trying again..
Jenkins execute full downstream build

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/BatchIndex.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    public void run(final Path root,
+                    final Runnable callback) {
+        try {
+            if (root == null) {
+                return;
+            }
+
+            logInformation("Starting indexing of " + root.toUri() + " ...");
```
Comment (jhrcek) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/BatchIndex.java]: @mbarkley I confirm that indexing now seems to work correctly both when cloning via REST and in UI as well. There's however one issue that this is showing bogus URLs in server.log when cloning via REST:

```
[org.uberfire.ext.metadata.io.BatchIndex] (Thread-155) Starting indexing of default://master@myspace/null/ ...
[org.uberfire.ext.metadata.io.BatchIndex] (Thread-155) Completed indexing of default://master@myspace/null/
[org.uberfire.ext.metadata.io.BatchIndex] (Thread-161) Starting indexing of default://master@myspace/null-1/ ...
[org.uberfire.ext.metadata.io.BatchIndex] (Thread-155) Starting indexing of default://master@myspace/null/ ...
```

This looks like minor issue which shouln't hold merging of this important fix. Should I file that separately of can that be fixed quickly?

Comment (mbarkley) [reply uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/BatchIndex.java]: @jhrcek Can you please open a new issue for this?

**Comment by mbarkley** [comment]
I don't think the test failures in the downstream build are related as the downstream build passes in the related PR.

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/BatchIndex.java (hunk 1 above)
Comment (jhrcek) [reply uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/BatchIndex.java]: @mbarkley My bad, I just noticed I didn't provide "name" argument in the CloneRepositoryRequest. When I fixed that, the repo is named correctly and everything is working fine. So I'm not going to report anything related to that. Sorry for confusion.

**Pull request merged**